Infrastructure/DataImporterService.cs

```diff
@@ -322,34 +338,77 @@ private void DeactivateSchools(NtpDbContext dbContext, List<int> importedSchoolU
 
                     await dbContext.SaveChangesAsync(cancellationToken);
 
-                    _logger.LogInformation("Updated Tuition Partner {TuitionPartnerName} with id of {TuitionPartnerId} from file {OriginalFilename}",
-                        existingTP.Name, existingTP.Id, originalFilename);
+                    AddToProcessedTPs("Updated", tuitionPartnerToProcess);
                 }
                 else
                 {
-                    _logger.LogInformation("No changes for Tuition Partner {TuitionPartnerName} with id of {TuitionPartnerId} from file {OriginalFilename}",
-                        existingTP.Name, existingTP.Id, originalFilename);
+                    AddToProcessedTPs(NoChangesLabel, tuitionPartnerToProcess);
                 }
             }
             else if (matchedTPs.Count > 1)
             {
                 var errorMsg = $"The file {originalFilename} with seo url('{tuitionPartnerToProcess.SeoUrl}') and import id('{tuitionPartnerToProcess.ImportId}') is returning more than 1 result from the database.";
-                _logger.LogError(message: errorMsg);
+                _logger.LogError(message: $"{StringConstants.TPDataImportLogMessagePrefix}{errorMsg}");
                 throw new InvalidOperationException(errorMsg);
             }
+        }
+
+        await DeactivateTPs(dbContext, allExistingTPs, cancellationToken);
 
-            successfullyProcessed.Add(tuitionPartnerToProcess.SeoUrl, tuitionPartnerToProcess);
+        return originalFilename;
+    }
+
+    private void LogProcessedTPsMessage(string filename)
+    {
+        var msg = $"The {filename} was successfully processed, with the following Tuition Partner outcomes.";
+        var hasChanges = false;
+
+        foreach (var kvp in _processedTPs)
+        {
+            if (kvp.Key != NoChangesLabel)
+                hasChanges = true;
+
+            msg += $"{Environment.NewLine}{Environment.NewLine}{Environment.NewLine}{kvp.Key}:{Environment.NewLine}{Environment.NewLine}";
```

Comment:
Instead of using `+=` for string concatenation in loops, consider using `StringBuilder` as it's more efficient for building large strings.

Reply:
Done